S/SEAL/build_tarballs.jl

```diff
@@ -3,65 +3,107 @@
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+    -DSEAL_BUILD_SEAL_C=ON \
+    -DSEAL_USE___BUILTIN_CLZLL=OFF \
+    -DSEAL___BUILTIN_CLZLL_FOUND_EXITCODE=1 \
+    -DSEAL___BUILTIN_CLZLL_FOUND_EXITCODE__TRYRUN_OUTPUT=1 \
+    -DSEAL_USE__ADDCARRY_U64=OFF \
+    -DSEAL__ADDCARRY_U64_FOUND_EXITCODE=1 \
+    -DSEAL__ADDCARRY_U64_FOUND_EXITCODE__TRYRUN_OUTPUT=1 \
+    -DSEAL_USE__SUBBORROW_U64=OFF \
+    -DSEAL__SUBBORROW_U64_FOUND_EXITCODE=1 \
+    -DSEAL__SUBBORROW_U64_FOUND_EXITCODE__TRYRUN_OUTPUT=1 \
+    -DSEAL_MEMSET_S_FOUND_EXITCODE=1 \
+    -DSEAL_MEMSET_S_FOUND_EXITCODE__TRYRUN_OUTPUT=1
+elif [[ "${target}" == aarch64* ]]; then
+  cmake -S . -B build \
+    -DCMAKE_INSTALL_PREFIX=$prefix \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN%.*}_clang.cmake \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+    -DSEAL_BUILD_SEAL_C=ON \
+    -DSEAL_USE___BUILTIN_CLZLL=OFF \
+    -DSEAL___BUILTIN_CLZLL_FOUND_EXITCODE=1 \
+    -DSEAL___BUILTIN_CLZLL_FOUND_EXITCODE__TRYRUN_OUTPUT=1 \
+    -DSEAL_USE__ADDCARRY_U64=OFF \
+    -DSEAL__ADDCARRY_U64_FOUND_EXITCODE=1 \
+    -DSEAL__ADDCARRY_U64_FOUND_EXITCODE__TRYRUN_OUTPUT=1 \
+    -DSEAL_USE__SUBBORROW_U64=OFF \
+    -DSEAL__SUBBORROW_U64_FOUND_EXITCODE=1 \
+    -DSEAL__SUBBORROW_U64_FOUND_EXITCODE__TRYRUN_OUTPUT=1 \
+    -DSEAL_ARM64_EXITCODE=1 \
+    -DSEAL_ARM64_EXITCODE__TRYRUN_OUTPUT=1
 else
-  cmake . \
+  cmake -S . -B build \
     -DCMAKE_INSTALL_PREFIX=$prefix \
     -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN%.*}_clang.cmake \
     -DCMAKE_BUILD_TYPE=Release \
     -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
-    -DBUILD_SHARED_LIBS=ON \
     -DSEAL_BUILD_SEAL_C=ON \
     -DSEAL_USE___BUILTIN_CLZLL=OFF \
+    -DSEAL___BUILTIN_CLZLL_FOUND_EXITCODE=1 \
+    -DSEAL___BUILTIN_CLZLL_FOUND_EXITCODE__TRYRUN_OUTPUT=1 \
     -DSEAL_USE__ADDCARRY_U64=OFF \
-    -DSEAL_USE__SUBBORROW_U64=OFF
+    -DSEAL__ADDCARRY_U64_FOUND_EXITCODE=1 \
+    -DSEAL__ADDCARRY_U64_FOUND_EXITCODE__TRYRUN_OUTPUT=1 \
+    -DSEAL_USE__SUBBORROW_U64=OFF \
+    -DSEAL__SUBBORROW_U64_FOUND_EXITCODE=1 \
+    -DSEAL__SUBBORROW_U64_FOUND_EXITCODE__TRYRUN_OUTPUT=1
 fi
-make -j${nproc}
-make install
+cmake --build build --parallel ${nproc}
+cmake --install build
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = [
-    Platform("x86_64", "linux"; libc="musl"),
-    Platform("x86_64", "macos"),
-    Platform("x86_64", "linux"; libc="glibc"),
-    Platform("aarch64", "linux"; libc="glibc"),
-    Platform("x86_64", "freebsd"),
-    Platform("powerpc64le", "linux"; libc="glibc"),
-    Platform("aarch64", "linux"; libc="musl")
+    Linux(:x86_64; libc=:glibc),
+    Linux(:x86_64; libc=:musl),
+    Linux(:aarch64; libc=:glibc),
+    Linux(:aarch64; libc=:musl),
+    Linux(:powerpc64le; libc=:glibc),
+    MacOS(:x86_64),
+    FreeBSD(:x86_64)
```

Comment:
This change should be reverted :slightly_smiling_face:

Reply:
OK, but why? If I locally run both codes, I get the following:
```julia
julia> using BinaryBuilder

julia> Linux(:x86_64, libc=:musl)
Linux(:x86_64, libc=:musl)

julia> Platform("x86_64", "linux"; libc="musl")
ERROR: MethodError: no constructors have been defined for Platform
Stacktrace:
 [1] top-level scope at REPL[6]:1
```
What am I missing here? And why does the `build_tarballs.jl` script work locally but not in the Yggdrasil CI?

Reply:
Here we use [Julia v1.6.0-rc1](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=9188&view=logs&j=96c13d9d-dfd2-51e5-2f76-4a0b3ce09abc&t=09532434-8b0a-5f0e-118d-a3181cde5ec8&l=14) + `BinaryBuilderBase#master` + `BinaryBuilder#master`.  The new platforms types don't exist in the latest stable version of `BinaryBuilder` because they've been introduced in Julia v1.6

Reply:
Aha! Thanks for clearing this up! Hopefully fixed in 3f523a9a.